From 14310a64cfe49248e853f9f02fb134a26ab4e22e Mon Sep 17 00:00:00 2001
From: Chitti Babu Theegala <ctheeg@codeaurora.org>
Date: Tue, 10 Apr 2018 12:47:40 +0530
Subject: [PATCH 7/9] sepolicy: Label vendor.post_boot.parsed

* Post boot script sets this property to indicate perfd is ready to operate

Change-Id: I34607c9d18949e88fa072c7f90862d7fdfe4a4d2
---
 common/init_shell.te     | 2 ++
 common/property.te       | 1 +
 common/property_contexts | 1 +
 common/system_server.te  | 2 ++
 4 files changed, 6 insertions(+)

diff --git a/common/init_shell.te b/common/init_shell.te
index 616a561..45d0627 100644
--- a/common/init_shell.te
+++ b/common/init_shell.te
@@ -59,6 +59,7 @@ allow qti_init_shell self:capability {
 # freq_prop - for setting frequency from postboot script
 # perfd_prop - for setting ctl.perfd property from postboot script
 # mpdecision_prop - for setting ctl.mpdecision property from postboot script
+# vendor_mpctl_prop - for setting vendor.post_boot.parsed property from postboot script
 # bluetooth_prop - for setting bt related properties from postboot script
 # uicc_prop - for access to UICC property
 # ctl_qmuxd_prop/ctl_netmgrd_prop - Needed in order to set properties on qmuxd and netmgrd processes
@@ -70,6 +71,7 @@ set_prop(qti_init_shell, freq_prop)
 set_prop(qti_init_shell, perfd_prop)
 set_prop(qti_init_shell, gamed_prop)
 set_prop(qti_init_shell, mpdecision_prop)
+set_prop(qti_init_shell, vendor_mpctl_prop)
 set_prop(qti_init_shell, bluetooth_prop)
 set_prop(qti_init_shell, config_prop)
 set_prop(qti_init_shell, sensors_prop)
diff --git a/common/property.te b/common/property.te
index 8ebc0a0..165ea31 100644
--- a/common/property.te
+++ b/common/property.te
@@ -55,6 +55,7 @@ type opengles_prop, property_type;
 type mdm_helper_prop, property_type;
 type mpdecision_prop, property_type;
 type gamed_prop, property_type;
+type vendor_mpctl_prop, property_type;
 
 #Needed for  ubwc support
 type debug_gralloc_prop, property_type;
diff --git a/common/property_contexts b/common/property_contexts
index e91270a..0c8f37f 100644
--- a/common/property_contexts
+++ b/common/property_contexts
@@ -65,6 +65,7 @@ ro.qualcomm.bt.hci_transport  u:object_r:bluetooth_prop:s0
 ctl.mdm_helper             u:object_r:mdm_helper_prop:s0
 ctl.mpdecision             u:object_r:mpdecision_prop:s0
 qualcomm.perf.cores_online u:object_r:mpdecision_prop:s0
+vendor.post_boot.parsed    u:object_r:vendor_mpctl_prop:s0
 netd.fstman.               u:object_r:netd_prop:s0
 location.                  u:object_r:location_prop:s0
 qc.izat.                   u:object_r:location_prop:s0
diff --git a/common/system_server.te b/common/system_server.te
index 905f8c0..0d9595a 100644
--- a/common/system_server.te
+++ b/common/system_server.te
@@ -171,6 +171,8 @@ allow system_server bservice_prop:file r_file_perms;
 allow system_server reschedule_service_prop:file r_file_perms;
 allow system_server appdomain:file w_file_perms;
 
+get_prop(system_server, vendor_mpctl_prop)
+
 # IPC call for sensor feed
 binder_call(system_server, mm-pp-daemon)
 binder_call(system_server, hal_camera)
-- 
2.17.1

